packages/astro/src/transitions/router.ts

```diff
@@ -637,8 +637,28 @@ if (inBrowser) {
 		originalLocation = new URL(location.href);
 		addEventListener('popstate', onPopState);
 		addEventListener('load', onPageLoad);
-		if ('onscrollend' in window) addEventListener('scrollend', onScroll);
-		else addEventListener('scroll', throttle(onScroll, 350), { passive: true });
+		if ('onscrollend' in window) addEventListener('scrollend', onScrollEnd);
+		// HACK: emulate "onscrollend" behavior by checking scroll positions with
+		// `requestAnimationFrame` every 350ms.
+		else {
+			let timestamp: number, y = scrollY, x = scrollX;
+			const cb = (now: number) => {
+				timestamp ??= now;
+				if (
+					now - timestamp > 350
+					&& y === scrollY && x === scrollX
+					&& (y !== history.state.scrollY || x !== history.state.scrollX)
+				) {
+					timestamp = now;
+					onScrollEnd();
+				}
+				// Save the current scroll positions.
+				y = scrollY, x = scrollX;
+				requestAnimationFrame(cb);
```

Comment:
Once started on the first page load, this runs forever on each frame?
Even when the browser throttles down animation frames due to inactivity, a new animation frame might well occur 60, 120 or even 240 times a second.